common/authorization/authorizationInterceptor.go

```diff
@@ -0,0 +1,109 @@
+// The MIT License
+//
+// Copyright (c) 2020 Temporal Technologies Inc.  All rights reserved.
+//
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package authorization
+
+import (
+	"context"
+	"strings"
+
+	"go.temporal.io/api/serviceerror"
+	"google.golang.org/grpc"
+
+	"go.temporal.io/server/common/metrics"
+)
+
+var (
+	errUnauthorized = serviceerror.NewPermissionDenied("Request unauthorized.")
+)
+
+func (a *interceptor) Interceptor(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler,
+) (interface{}, error) {
+
+	var ns string
+	rwns, ok := req.(RequestWithNamespace)
+	if ok {
+		ns = rwns.GetNamespace()
+	}
+
+	apiName := info.FullMethod
+	index := strings.LastIndex(apiName, "/")
+	if index > -1 {
```

Comment:
do we expect if info.FullMethod does not contain a `/` ?

Reply:
They come in the of `/temporal.api.workflowservice.v1.WorkflowService/DescribeNamespace`. So a slash is 100% expected. Even if somehow it's not there, we'll use the full string as the API name.